### PR TITLE
Mail notification cache

### PR DESCRIPTION
**Pull request by legionth**
Every new instance of `ilForumAuthorInformation` will call several methods to create the author information. To avoid this a cache object will be created, that can be used instead of creating new instances.

### CODE REVIEW

**Comment by legionth** [comment]
Ping @mjansenDatabay

File: Modules/Forum/classes/class.ilForumCronNotification.php [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 include_once "Services/Cron/classes/class.ilCronJob.php";
 include_once "./Modules/Forum/classes/class.ilForumMailNotification.php";
+require_once './Modules/Forum/classes/class.ilForumNotificationCache.php';
```
Comment (mjansenDatabay) [inline Modules/Forum/classes/class.ilForumCronNotification.php]: Please update the composer class map instead.

File: Modules/Forum/classes/class.ilForumCronNotificationDataProvider.php [hunk 2]
```diff
@@ -4,6 +4,7 @@
 include_once './Modules/Forum/interfaces/interface.ilForumNotificationMailData.php';
 include_once './Modules/Forum/classes/class.ilForumProperties.php';
 require_once 'Modules/Forum/classes/class.ilForumAuthorInformation.php';
+require_once './Modules/Forum/classes/class.ilForumNotificationCache.php';
```
Comment (mjansenDatabay) [inline Modules/Forum/classes/class.ilForumCronNotificationDataProvider.php]: Please update the composer class map instead.

File: Modules/Forum/classes/class.ilForumNotificationCache.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<?php
+
```
Comment (mjansenDatabay) [inline Modules/Forum/classes/class.ilForumNotificationCache.php]: Missing ILIAS license information (similar to other ILIAS class files)...

File: Modules/Forum/classes/class.ilForumNotificationCache.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+class ilForumNotificationCache
```
Comment (mjansenDatabay) [inline Modules/Forum/classes/class.ilForumNotificationCache.php]: Missing PhpDoc (@author) ...

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

**Comment by legionth** [comment]
Updated the branch based on the remarks. Have a look.

**Pull request merged**